### DISCUSSION TITLE
Fallback/Secondary SMTP Service Definition

### DISCUSSION THREAD

**Discussion by cormip** [meteor/meteor#12519] [Feature Requests]
After dealing with today's Mailgun API outage, I thought it'd be very useful to be able to configure a fallback SMTP service that would automatically take over when the primary SMTP service stops working for whatever reason. I know I could likely rig something up by catching errors and leveraging the existing Meteor Email API, but it'd be so much easier to simply configure (or something similar):

```bash
MAIL_URL=smtp://user:pass@primary-service.com:587
MAIL_URL_ALT_1=smtp://user:pass@alternate-service.com:587
MAIL_URL_ALT_2=smtp://user:pass@last-hope-service.com:587
```

I recall seeing this capability on other application server platforms I've used in the past.

**Comment by StorytellerCZ** [comment]
I was just talking the other day with @jnv from Superface.ai about creating a simple package for Meteor to do just this via Superface. No promises, but we'll try to get to it asap.
That said I don't think that it is something for Meteor core to worry about.

**Reply by jnv** [reply]
Thanks for the mention! Yeah, we support email failover, in fact @StorytellerCZ wrote how to use our SDK with Meteor's custom email transport, but I suspect it's outdated: https://dev.to/storytellercz/resilient-e-mail-sending-in-meteor-with-superface-ai-mo1

I'm just not sure if it should be a complementary or alternative package to Meteor's official `email` package.

**Reply by StorytellerCZ** [reply]
I think we can make it complimentary by using the send email hook. That way we won't break any dependencies.
